src/lib/SX1280Driver/SX1280.cpp

```diff
@@ -0,0 +1,344 @@
+{
+    hal.init();
+    hal.TXdoneCallback = &SX1280Driver::TXnbISR;
+    hal.RXdoneCallback = &SX1280Driver::RXnbISR;
+
+    hal.reset();
+    Serial.println("SX1280 Begin");
+    delay(100);
+
+    uint16_t firmwareRev = (((hal.ReadRegister(REG_LR_FIRMWARE_VERSION_MSB)) << 8) | (hal.ReadRegister(REG_LR_FIRMWARE_VERSION_MSB + 1)));
+    Serial.print("Firmware Revision: ");
+    Serial.println(firmwareRev);
+
+    this->SetMode(SX1280_MODE_STDBY_RC);                                    //step 1 put in STDBY_RC mode
+    hal.WriteCommand(SX1280_RADIO_SET_PACKETTYPE, SX1280_PACKET_TYPE_LORA); //Step 2: set packet type to LoRa
+    this->ConfigModParams(currBW, currSF, currCR);                          //Step 5: Configure Modulation Params
+    hal.WriteCommand(SX1280_RADIO_SET_AUTOFS, 0x01);                        //enable auto FS
+    this->SetPacketParams(12, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, SX1280_LORA_IQ_NORMAL);
+    this->SetFrequency(this->currFreq); //Step 3: Set Freq
+    this->SetFIFOaddr(0x00, 0x00);      //Step 4: Config FIFO addr
+    this->SetDioIrqParams(SX1280_IRQ_RADIO_ALL, SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE); //
+}
+
+void ICACHE_RAM_ATTR SX1280Driver::Config(SX1280_RadioLoRaBandwidths_t bw, SX1280_RadioLoRaSpreadingFactors_t sf, SX1280_RadioLoRaCodingRates_t cr, uint32_t freq, uint8_t PreambleLength)
+{
+    this->SetMode(SX1280_MODE_STDBY_XOSC);
+    ConfigModParams(bw, sf, cr);
+    SetPacketParams(PreambleLength, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, SX1280_LORA_IQ_NORMAL); // TODO don't make static etc.
+    SetFrequency(freq);
+}
+
+void ICACHE_RAM_ATTR SX1280Driver::SetOutputPower(int8_t power)
+{
+    uint8_t buf[2];
+    buf[0] = power + 18;
+    buf[1] = (uint8_t)SX1280_RADIO_RAMP_04_US;
+    hal.WriteCommand(SX1280_RADIO_SET_TXPARAMS, buf, 2);
+    Serial.print("SetPower: ");
+    Serial.println(buf[0]);
+    return;
+}
+
+void SX1280Driver::SetPacketParams(uint8_t PreambleLength, SX1280_RadioLoRaPacketLengthsModes_t HeaderType, uint8_t PayloadLength, SX1280_RadioLoRaCrcModes_t crc, SX1280_RadioLoRaIQModes_t InvertIQ)
+{
+    uint8_t buf[7];
+
+    buf[0] = PreambleLength;
+    buf[1] = HeaderType;
+    buf[2] = PayloadLength;
+    buf[3] = crc;
+    buf[4] = InvertIQ;
+    buf[5] = 0x00;
+    buf[6] = 0x00;
+
+    hal.WriteCommand(SX1280_RADIO_SET_PACKETPARAMS, buf, sizeof(buf));
+}
+
+void SX1280Driver::SetMode(SX1280_RadioOperatingModes_t OPmode)
+{
+
+    //if (OPmode == currOpmode)
+    //{
+     //   return;
+    //}
+
+    WORD_ALIGNED_ATTR uint8_t buf3[3]; //TODO make word alignmed
+
+    switch (OPmode)
+    {
+
+    case SX1280_MODE_SLEEP:
+        hal.WriteCommand(SX1280_RADIO_SET_SLEEP, 0x01);
+        break;
+
+    case SX1280_MODE_CALIBRATION:
+        break;
+
+    case SX1280_MODE_STDBY_RC:
+        hal.WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_RC);
+        break;
+    case SX1280_MODE_STDBY_XOSC:
```

Comment:
Would be nice to keep spacing consistent here (i.e. line return between the new case)